sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -848,15 +856,39 @@ export class MessageReceiver extends LinkEntity {
 
   /**
    * Will reconnect the receiver link if necessary.
-   * @param {AmqpError | Error} [receiverError] The receiver error if any.
+   * @param receiverError The receiver error or connection error, if any.
+   * @param connectionDidDisconnect Whether this method is called as a result of a connection disconnect.
    * @returns {Promise<void>} Promise<void>.
    */
-  async onDetached(receiverError?: AmqpError | Error): Promise<void> {
+  async onDetached(receiverError?: AmqpError | Error, causedByDisconnect?: boolean): Promise<void> {
     const connectionId = this._context.namespace.connectionId;
+
+    // User explicitly called `close` on the receiver, so link is already closed
+    // and we can exit early.
+    if (this.wasCloseInitiated) {
+      log.error(
+        `[${connectionId}] Receiver '${this.name}' with address '${this.address}' has had onDetached called after being explictly closed.`
+      );
+      return;
+    }
+
+    // Prevent multiple onDetached invocations from running concurrently.
+    if (this._isDetaching) {
+      // This can happen when the network connection goes down for some amount of time.
+      // The first connection `disconnect` will trigger `onDetached` and attempt to retry
+      // creating the connection/receiver link.
+      // While those retry attempts fail (until the network connection comes back up),
+      // we'll continue to see connection `disconnect` errors.
+      // These should be ignored until the already running `onDetached` completes
+      // its retry attempts or errors.
+      log.error(
+        `[${connectionId}] Receiver '${this.name}' with address '${this.address}' has had onDetached called while already detaching.`
```

Comment:
The log statement preceding this would be `<connectionId> calling detached on streaming receiver '<receiver name>'`. Can we stay in sync with that pattern to help with readability?

```suggestion
        `[${connectionId}] Call to detached is already in progress for streaming receiver '${this.name}'.`
```

Reply:
Sure. I admittedly did this to match the pattern used in this method, but I get that in the overall flow of logs your suggestion may be easier to read.